pkg/nfd/config.go

```diff
@@ -0,0 +1,58 @@
+package nfd
+
+import (
+	. "github.com/rh-ecosystem-edge/nvidia-ci/pkg/global"
+	"time"
+)
+
+type CustomConfig struct {
+	CustomCatalogSourceIndexImage string
+	CreateCustomCatalogsource     bool
+
+	CustomCatalogSource string
+	CatalogSource       string
+	CleanupAfterInstall bool
+}
+
+// NewCustomConfig creates a new CustomConfig instance with default settings.
+// All string fields are initialized to UndefinedValue and boolean fields to false.
+func NewCustomConfig() *CustomConfig {
+	return &CustomConfig{
+		CustomCatalogSourceIndexImage: UndefinedValue,
+		CreateCustomCatalogsource:     false,
+
+		CustomCatalogSource: UndefinedValue,
+		CatalogSource:       UndefinedValue,
+		CleanupAfterInstall: false,
+	}
+}
```

Comment:
It looks like this can be shared by all operators that might need a custom catalog source, not limited to NFD.